lib/iris/_lazy_data.py

```diff
@@ -146,6 +153,25 @@ def _optimum_chunksize(chunks, shape, limit=None, dtype=np.dtype("f4")):
     return tuple(result)
 
 
+@wraps(_optimum_chunksize_internals)
+def _optimum_chunksize(
+    chunks,
+    shape,
+    limit=None,
+    dtype=np.dtype("f4"),
+):
+    # By providing dask_array_chunksize as an argument, we make it so that the
+    # output of _optimum_chunksize_internals depends only on its arguments (and
+    # thus we can use lru_cache)
+    return _optimum_chunksize_internals(
+        tuple(chunks),
+        tuple(shape),
+        limit=limit,
+        dtype=dtype,
+        dask_array_chunksize=dask.config.get("array.chunk-size"),
```

Comment:
Personally, I'd move the 'dask.utils.parse_bytes' call here, too, to put all the dask-specific bits in one place.

Reply:
Can that have a result that changes between calls? I thought I'd checked and it can't.

Reply:
> Can that have a result that changes between calls?

No I dont think so.
I just thought that it makes good general sense to keep all the dask-specific code here, remove that dependency from the underlying utility routine.

Reply:
I don't really mind either way - this way is presumably slightly faster and the change would be a bit cleaner